package/utils/phicommk3-screenctrl/Makefile

```diff
@@ -0,0 +1,61 @@
+include $(TOPDIR)/rules.mk
+
+PKG_NAME:=phicommk3-screenctrl
+PKG_VERSION:=1.0
+PKG_RELEASE:=2
+
+PKG_SOURCE_PROTO:=git
+PKG_SOURCE_URL:=https://github.com/Hill-98/k3screenctrl
+PKG_SOURCE_VERSION:=86c4ebc227212bcc407175906686c2c8c70c7b61
+PKG_SOURCE_SUBDIR:=$(PKG_NAME)-$(PKG_SOURCE_VERSION)
+PKG_SOURCE:=$(PKG_SOURCE_SUBDIR).tar.xz
+PKG_BUILD_DIR:=$(BUILD_DIR)/$(PKG_SOURCE_SUBDIR)
+
+PKG_MAINTAINER:=Hamster Tian <haotia@gmail.com>
+
+include $(INCLUDE_DIR)/package.mk
+
+TARGET_CFLAGS+= -D_GNU_SOURCE
+
+define Package/$(PKG_NAME)
+  SECTION:=utils
+  CATEGORY:=Utilities
+  DEPENDS:=@TARGET_bcm53xx_DEVICE_phicomm-k3 +@BUSYBOX_CONFIG_ARPING +@KERNEL_DEVMEM +bash
```

Comment:
I do not like the dependencies.
In the normal build done by build bot we will probably not have BUSYBOX_CONFIG_ARPING and KERNEL_DEVMEM activated. 
I do not know this package in detail, has someone a better solution?

Reply:
The k3screenctrl is came from @updateing Aka Tian Hao <haotia@gmail.com> , I will try to contact him for this.

Reply:
The screen is connected to second serial port on bcm53xx. However setting status to okay in DTS does not suffice in activating the port - we have to change several bits in DMU registers (code in [Broadcom SDK](https://github.com/RMerl/asuswrt-merlin/blob/c2fa8fd364594d2461af4bc26cc83455868cc6c0/release/src-rt-7.14.114.x/src/linux/linux-2.6.36/arch/arm/plat-brcm/bcm5301x_dmu.c#L533) and [mine](https://github.com/updateing/k3screenctrl/blob/master/src/main.c#L58)). I could not find any way to control the bits in DTS, so I chose to modify that via direct memory access, and thus `KERNEL_DEVMEM`.

`arping` is not required by k3screenctrl. Probably @Hill-98 knows it better?

Reply:
He is not a coder either

Reply:
arping is the device I use to determine if the device connected to the router is online so that it is displayed on the fifth page of the screen. Specific actions: A shell script monitors the DHCP list every minute, then uses arping to test whether these devices can communicate, and if so, the device is online. I know this is a stupid way, I am not a professional developer, if you have a better way to replace it, or use the ping command, and do not need arping.

Reply:
Associated file: ```/lib/k3screenctrldevice_online.sh```